src/Reflection/ClassReflection.php

```diff
@@ -453,6 +467,15 @@ public function getMethod(string $methodName, ClassMemberAccessAnswerer $scope):
 		if ($scope->isInClass()) {
 			$key = sprintf('%s-%s', $key, $scope->getClassReflection()->getCacheKey());
 		}
+
+		if ($this->requireExtendsMethodsClassReflectionExtension->hasMethod($this, $methodName)) {
```

Comment:
Maybe do it below the foreach? If the method already exists the native one should have precedence.